e2e_playwright/st_rerun_test.py

```diff
@@ -47,6 +47,8 @@ def test_fragment_scoped_st_rerun(app: Page):
     expect(app.get_by_test_id("stMarkdown").nth(1)).to_have_text(
         "fragment run count: 10"
     )
+    # wait for app run to make sure that we do not check a stale state of the app
+    wait_for_app_run(app)
```

Comment:
nit: I'm not sure if this is necessary since the `click_button` call already waits for the rerun.

Reply:
Hmhm good point. Then I guess it was a race condition that made the test from the original PR pass. In this case, the best thing probably is to click a couple of more times on the button to increase the chance of the race to be triggered (likelihood should be fairly high given how often the test failed _after_ the PR was merged 😅)